powerfactory/powerfactory-converter/src/main/java/com/powsybl/powerfactory/converter/NodeConverter.java

```diff
@@ -0,0 +1,170 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.powerfactory.converter;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import org.apache.commons.lang3.mutable.MutableInt;
+
+import com.google.common.primitives.Ints;
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.network.Substation;
+import com.powsybl.iidm.network.TopologyKind;
+import com.powsybl.iidm.network.VoltageLevel;
+import com.powsybl.powerfactory.converter.PowerFactoryImporter.ImportContext;
+import com.powsybl.powerfactory.converter.PowerFactoryImporter.NodeRef;
+import com.powsybl.powerfactory.model.DataObject;
+import com.powsybl.powerfactory.model.DataObjectRef;
+
+/**
+ * @author Luma Zamarreño <zamarrenolm at aia.es>
+ * @author José Antonio Marqués <marquesja at aia.es>
+ */
+
+class NodeConverter extends AbstractConverter {
+
+    NodeConverter(ImportContext importContext, Network network) {
+        super(importContext, network);
+    }
+
+    /**
+     * A cubicle defines the connection between a node and a side of an element.
+     * A staSwitch object is automatically created inside the cubicle when an element is connected to a node.
+     * An ElmCoup is also connected to a node through a cubicle, but without any additional StaSwitch.
+     *
+     * An iidm node is created by each elmTerm and by each staSwitch connected to an element.
+     */
+    void createAndMapConnectedObjs(DataObject elmTerm) {
+        VoltageLevel vl = findVoltageLevel(getNetwork(), getImportContext(), elmTerm);
+
+        // Create the node defined by elmTerm
+        int node = createNode(vl);
+
+        getImportContext().elmTermIdToNode.putIfAbsent(elmTerm.getId(), new NodeRef(vl.getId(), node, 0));
+
+        NodeModel nodeModel = NodeModel.create(vl, elmTerm);
+        createBusbarSectionIfNodeHasBeenDefinedAsSuch(vl, node, nodeModel);
+
+        // Create additional nodes associated to switches (staSwitch) included in cubicles
+        createAddtionalNodesAndMapConnectedObjs(vl, node, elmTerm);
+    }
+
+    private static VoltageLevel findVoltageLevel(Network network, ImportContext importContext, DataObject elmTerm) {
```

Comment:
can be made non-`static`, no need to pass network and import context as parameters. They can be accessed using `getNetwork()` and `getImportContext()`